util/java/libtiled-java/src/main/java/tiled/view/IsometricRenderer.java

```diff
@@ -0,0 +1,94 @@
+/*-
+ * #%L
+ * This file is part of libtiled-java.
+ * %%
+ * Copyright (C) 2004 - 2016 Thorbjørn Lindeijer <thorbjorn@lindeijer.nl>
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+package tiled.view;
+
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Image;
+import tiled.core.Map;
+import tiled.core.Tile;
+import tiled.core.TileLayer;
+
+/**
+ * The isometric map renderer.
+ *
+ * @author Mike Thomas
+ * @version 0.17
+ */
+public class IsometricRenderer implements MapRenderer {
+
+    private final Map map;
+
+    /**
+     * <p>Constructor for IsometricRenderer.</p>
+     *
+     * @param map a {@link tiled.core.Map} object.
+     */
+    public IsometricRenderer(Map map) {
+        this.map = map;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Dimension getMapSize() {
+        int side = map.getHeight() + map.getWidth();
+        return new Dimension(
+            side * map.getTileWidth() / 2,
+            side * map.getTileHeight() / 2);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void paintTileLayer(Graphics2D g, TileLayer layer) {
+        final int tileWidth = map.getTileWidth();
+        final int tileHeight = map.getTileHeight();
+
+        final int hOffset = (map.getWidth() * tileWidth) / 2;
+
+        for (int x = 0; x < layer.getWidth(); x++) {
+            for (int y = 0; y < layer.getHeight(); y++) {
```

Comment:
Tiled renders row by row, iterating diagonally over the 2D tiles array, but I think in most cases there will be no visual difference.